src/relay/pass/quantize.cc

```diff
@@ -738,6 +739,48 @@ TVM_REGISTER_API("relay._quantize.temp_expr_realize")
   return n->Realize();
 });
 
+// =============
+// calibration
+
+class StatsCollector : private ExprMutator {
```

Comment:
If just collect stats, `ExprVisitor` should be enough

Reply:
`ExprMutator` is needed actually. This mutator transform annotated expr to original expr by removing each `simulate_quantize`.
For example Relay program:
```
%1 = ..
%2 = simulate_quantize(%1)
%3 = op(%2)
%4 = simulate_quantize(%3)
```
We need to profile %1 and %3. But %3 takes %2 as input, we need to replace input of %3 with %1 (because in Annotate pass simulate_quantize in %2 is not in passthrough mode, we need to either remove it or rewrite it in passthrough mode)

Reply:
@ZihengJiang I was thinking that the other pr #3543 actually breaks this pass (because the result of this pass contains annotations and casts)

Reply:
Why not collect stats before annotate?

Reply:
@ZihengJiang Annotations tell us which nodes should be profiled. If we want to collect stats before annotate, we need to repeat the code similar to annotate to decide which node should be quantized.

Reply:
Okay, let's keep the current way.  #3543 will not breaks this pass since `annotation.cast_hint` and `annotation.stop_fusion` will not change the running result. They are just annotation and you can view them as `identity`. One thing is, instead of detecting and jumping `simulated_quantize` inside of `IRMutator`, let's adding an option like `simulated_quantize(kind=kIdentity)` for eliminating the impact of `simulated_quantize`

Reply:
@ZihengJiang updated